pkg/checkpoint/process_test.go

```diff
@@ -170,8 +166,9 @@ func TestProcess(t *testing.T) {
 				},
 				"AA": {},
 			},
-			expectStop:        []string{"AA", "kube-system/pod-checkpointer"},
-			expectGraceRemove: []string{"AA", "kube-system/pod-checkpointer"},
+			expectStart:       []string{"kube-system/pod-checkpointer"},
+			expectStop:        []string{"BB"},
+			expectGraceRemove: []string{"AA", "BB", "kube-system/pod-checkpointer"},
```

Comment:
There's a couple of these, but why would you expect a Start and GraceRemove for pod-checkpointer. And why would a particular pod have multiple states anyway?

Reply:
Start is because if we're not stopping the checkpointer, we're starting it (was true in the old code too, but now we have two phases--before and after grace period).

The state transitions are: None (implicit before the test starts) => Active Grace Period (the parent is removed, but keep active for a bit, which leads to a "start" per my previous comment) => Remove (grace period has passed)

I had to reason through the other test changes but they made sense to me too: we always start from "no" state. The fix in my second commit was realizing that the checkpointer needed to keep in active grace period (rather than inactive) when its parent is removed to make sure it can clean up all its state before finally self-destructing.